### DISCUSSION TITLE
New code not getting updated neither in "briefcase build android" nor through "briefcase run android"

### DISCUSSION THREAD

**Discussion by GnathMath** [beeware/briefcase#1959] [Q&A] [answered]
I have made a simple code which I give you below
```
class G_Quiz(toga.App):
    def startup(self):

        #STARTUP BOX
        startupBox = toga.Box()
        startupBox.style = Pack(background_color='blue')

        #main box
        self.main_box = toga.Box(style=Pack(direction=COLUMN))

        # text box
        name_label = toga.Label("Enter the text: ")
        name_label.style = Pack(flex=1,padding=5)
        self.name_input = toga.TextInput()
        self.name_input.style = Pack(flex=2,padding=5)
        name_box = toga.Box(style=Pack(direction=COLUMN, padding=5))


        # now add label to this box
        name_box.add(name_label)
        name_box.add(self.name_input)


        # adding first box and button in main box
        firstBox = toga.Box(style=Pack(direction=COLUMN, padding=5))

        # button
        button = toga.Button("Submit", on_press=self.say_hello, style=Pack(padding=5))

        firstBox.add(name_box)
        firstBox.add(button)
        self.main_box.add(firstBox)

        self.main_window = toga.MainWindow(title=self.formal_name)
        self.main_window.content = self.main_box
        self.main_window.show()

    def say_hello(self, widget):
        self.main_window.info_dialog("Message", "Hi " + self.name_input.value)
        

def main():
    return G_Quiz()
```
Later I have updated this
```
class G_Quiz(toga.App):
    def startup(self):
        #STARTUP BOX
        startupBox = toga.Box()
        startupBox.style = Pack(background_color='blue')

        #main box
        self.main_box = toga.Box(style=Pack(direction=COLUMN))

        # text box
        name_label = toga.Label("Enter the text: ")
        name_label.style = Pack(flex=1,padding=5)
        self.name_input = toga.TextInput()
        self.name_input.style = Pack(flex=2,padding=5)
        name_box = toga.Box(style=Pack(direction=COLUMN, padding=5))


        # now add label to this box
        name_box.add(name_label)
        name_box.add(self.name_input)


        # adding first box and button in main box
        firstBox = toga.Box(style=Pack(direction=COLUMN, padding=5))

        # button
        button = toga.Button("Submit", on_press=partial(self.newBoxNavigation, data=firstBox), style=Pack(padding=5))

        firstBox.add(name_box)
        firstBox.add(button)
        self.main_box.add(firstBox)

        # main_box.add(view)

        self.main_window = toga.MainWindow(title=self.formal_name)
        self.main_window.content = self.main_box
        self.main_window.show()

    def newBoxNavigation(self, widget, data):
        newBox = toga.Box(style=Pack(direction=ROW))
        newLabel = toga.Label("Hi " + self.name_input.value, style=Pack(padding=(0, 5)))
        self.name_input.value = None
        if not os.path.exists(os.path.join('.','Files')):
            os.mkdir("Files")
        else:
            print('Files folder already exists')
        address = toga.Label(os.path.abspath(os.curdir))
        t = datetime.datetime.now()
        timeLabel = toga.Label(t)

        newBox.add(newLabel)
        newBox.add(address)
        newBox.add(timeLabel)
        self.main_box.add(newBox)

    def say_hello(self, widget, data, d2):
        t = datetime.datetime.now()
        newBox = toga.Box(style=Pack(direction=ROW))
        newLabel = toga.Label("Hi " + self.name_input.value + 'and current time is '+ str(t), style=Pack(padding=(0, 5)))
        if not os.path.exists(os.path.join('.','Files')):
            os.mkdir("Files")
        else:
            print('Files folder already exists')
        address = toga.Label(os.path.abspath(os.curdir)+'\n'+str(os.listdir(os.curdir)))
        address.style = Pack(width=200)

        newBox.add(newLabel)
        newBox.add(address)
        self.window2 = toga.Window(title="Message")
        self.window2.content = newBox
        self.window2.show()
        # self.main_window.info_dialog("Message", "Hi " + self.name_input.value)
        # data.add(d2)


def main():
    return G_Quiz()
```
Both worked fine in developer mode. I have installed the first code using "briefcase run android" using USB. It worked, but later when I updated the code and installed the same, it did not work. 

I have tried "briefcase update", "briefcase run -u", I have uninstalled the App and then tried to reinstall from the beginning. I still got the App which is stuck with the old code and new code functions are not getting updated. Can someone help me?

**Comment by freakboy3742** [comment]
You'll have to clarify what you mean by "does not work".

The ticket title suggests that you think the issue is that the code isn't being updated - is that *actually* the problem you're seeing? Are you getting an error? If so, what is it? 

If you're not seeing an error, just not seeing changes being reflect - How are you establishing that the code isn't being updated? Are you *certain* that you've saved the file locally? Can you provide a build log (add `--log` to the briefcase command) verifying that the code is being updated?

**Reply by GnathMath** [reply]
First of all, thank you so much for the timely reply. What I meant is my App is working in Android based on my old code, not with my new updated code. And the strange thing is that I do not get any error. It says BUILD SUCCESSFUL when I use a command "briefcase build android". I know the code is not being updated because I have updated the App through USB debugging and updated App in Android still functions with the old code. I have also uploaded the log file as you asked. 

I hope I clarified things correctly. If not, please tell me what else do I provide?
[briefcase.2024_08_14-12_55_05.build.log](https://github.com/user-attachments/files/16609067/briefcase.2024_08_14-12_55_05.build.log)

**Comment by GnathMath** [comment]
I have tried myself several possibilities and found a weird temporary solution to my problem.

I have deleted the directory "android" which is situated in "PROJECT FOLDER/APP NAME FOLDER/build/APP NAME FOLDER" and then when I built the apk file and tested it. This time the code is actually getting updated. I have also tried deleting only the apk file which was situated in "PROJECT FOLDER/APP NAME FOLDER/build/APP NAME FOLDER/android/gradle/app/build/outputs/apk/debug", which did not actually update the code. But deleting the entire directory of 'android' did the trick. 

Still this is not the correct solution, but my app works now. 

I once again thank @freakboy3742 for his reply.

**Reply by rmartin16** [reply, marked as answer]
Just in case this is useful, when you run the `briefcase update` command or use the `-u` flag, you'll still need to specify the relevant platform. In this case, `briefcase update android` and `briefcase run android -u`. Otherwise, the Briefcase command will run for the platform of the host machine, Windows in this case, instead of Android.

**Reply by GnathMath** [reply]
Oh my God. That works. That is the permanent solution. I did not know this earlier. You just have saved my day.
